From 344e26850aa8154503055de86e7b1e449a26134a Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 31 Jul 2013 17:10:29 +0900
Subject: [PATCH 1062/1083] drm: Fix frame buffer resolution is set as HDMI

The resolution of a frame buffer is set as HDMI.
When HDMI is not connected, Full HD size is chosen.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/gpu/drm/drm_fb_helper.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/gpu/drm/drm_fb_helper.c b/drivers/gpu/drm/drm_fb_helper.c
index 78f1732..28e0db0 100644
--- a/drivers/gpu/drm/drm_fb_helper.c
+++ b/drivers/gpu/drm/drm_fb_helper.c
@@ -809,7 +809,15 @@ int drm_fb_helper_single_fb_probe(struct drm_fb_helper *fb_helper,
 		struct drm_display_mode *desired_mode;
 		desired_mode = fb_helper->crtc_info[i].desired_mode;
 
+#if defined(CONFIG_DRM_RCAR_DU)
+#if defined(CONFIG_MACH_LAGER)
+		if ((desired_mode) && (i == 0)) {
+#elif defined(CONFIG_MACH_KOELSCH)
+		if ((desired_mode) && (i == 1)) {
+#endif
+#else
 		if (desired_mode) {
+#endif
 			if (gamma_size == 0)
 				gamma_size = fb_helper->crtc_info[i].mode_set.crtc->gamma_size;
 			if (desired_mode->hdisplay < sizes.fb_width)
@@ -824,6 +832,13 @@ int drm_fb_helper_single_fb_probe(struct drm_fb_helper *fb_helper,
 		}
 	}
 
+#if defined(CONFIG_DRM_RCAR_DU)
+	if (crtc_count == 0) {
+		sizes.fb_width = sizes.surface_width = 1920;
+		sizes.fb_height = sizes.surface_height = 1080;
+		crtc_count++;
+	}
+#endif
 	if (crtc_count == 0 || sizes.fb_width == -1 || sizes.fb_height == -1) {
 		/* hmm everyone went away - assume VGA cable just fell out
 		   and will come back later. */
-- 
1.8.3.2

